File: frontend/src/types/activity_log.ts
/** Mirrors backend app/models/activity_log.py AdminActivityLogEntry. */
export interface AdminActivityLogEntry {
  id: string
  admin_id: string
  admin_username: string
  action: string
  method: string
  path: string
  detail: Record<string, string> | null
  created_at: string
}

File: backend/app/models/activity_log.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class AdminActivityLogEntry(BaseModel):
    id: UUID
    admin_id: UUID
    admin_username: str
    action: str
    method: str
    path: str
    detail: dict[str, str] | None = None
    created_at: datetime
